widgets/waterTracker/main.js

```diff
@@ -120,6 +152,24 @@ const removeWater = (id) => {
 
 const reset = () => {
     waterArray = [];
+    otherProps = {goal: 2000, time: 60};
+}
+
+const timerFunc = () => {
+    const lastRecordDate = new Date(waterArray[waterArray.length-1].time);
+    const now = new Date();
+    if(lastRecordDate.getDate() < now.getDate()){
+        waterArray = []
+    }
+    alertModal('Drink Water', `Drink some water. Drinking water at right time is good for your health.`);
```

Comment:
Now that I've waited to see the modal, I see that t was a title.  I found the reference in line 117.  My 't' always equals 'time' comes from my physics background.

Thanks again for getting me to learn.  I ended up spending the hour waiting for the modal reading about JS OOP and Bootstrap Modal.  Because I primarily learned JS OOP while learning React, my mind had put them together, forgetting that I really need to know it for Vanilla JS.

Reply:
I am really sorry for that. I don't really have a habit of writing long descriptive texts. I sometimes get lazy and write only the important part.